src/promo/promo.service.ts

```diff
@@ -8,52 +9,74 @@ export class PromocodeService {
   constructor(private promocodeRepository: PromocodeRepository) {}
 
   async getListOfPromocodes(pageOptions: IPageOptions): Promise<IPromoCode[]> {
-    const promocodes: IPromoCode[] =
+    try {
+      const promocodes: IPromoCode[] =
       await this.promocodeRepository.getListOfPromocodes(pageOptions);
-    return promocodes;
+      return promocodes;
+    } catch (e) {
+      throw new RpcException(e);
+    }
   }
 
   async addPromoCode(promocode: IAddPromoCode): Promise<IPromoCode> {
-    const newPromocode: IPromoCode =
-      await this.promocodeRepository.addPromoCode(promocode);
-    return newPromocode;
+    try {
+      const newPromocode: IPromoCode =
+        await this.promocodeRepository.addPromoCode(promocode);
+      return newPromocode;
+    } catch (e) {
+      throw new RpcException(e);
+    }
   }
 
   async removePromoCode(promocode: IRemovePromoCode): Promise<boolean> {
-    if (!(await this.promocodeRepository.removePromoCode(promocode))) {
-      throw new NotFoundException(
-          `The promocode ${promocode.name} not found.`,
-      );
+    try {
+      if (!(await this.promocodeRepository.removePromoCode(promocode))) {
+        throw new RpcException(`The promocode ${promocode.name} not found.`);
+      }
+      return true;
+    } catch (e) {
+      throw new RpcException(e);
```

Comment:
As far as i remember  error is type of RpcException, so we don't need to wrap it to RpcException

Reply:
If we use RpcException, then we get an error in a readable form, otherwise just an Internal server error